subprojects/language-native/src/integTest/groovy/org/gradle/language/cpp/CppIncrementalBuildIntegrationTest.groovy

```diff
@@ -693,4 +770,8 @@ class CppIncrementalBuildIntegrationTest extends AbstractCppInstalledToolChainIn
         then:
         nonSkippedTasks.empty
     }
+
+    private boolean unresolvedHeadersDetected() {
+        output.contains("After parsing the source files, Gradle cannot calculate the exact set of include files for dependDebugCpp. Every file in the include search path will be considered a header dependency.")
```

Comment:
Shouldn't we display the path of the task here?

Reply:
It would be cleaner if we passed the task name to this method instead of assuming that it's always called from a context where there is a `dependDebugCpp` task.

Reply:
Now we show the task path and I also added an argument to `unresolvedHeadersDetected`.